src/Nethermind/Nethermind.Network/P2P/ConcurrentPacketAcceptanceStrategy.cs

```diff
@@ -0,0 +1,26 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Network.Rlpx;
+
+namespace Nethermind.Network.P2P;
+
+public class ConcurrentPacketAcceptanceStrategy : IPacketAcceptanceStrategy
+{
+    private readonly IPacketAcceptanceStrategy _strategy;
+    private readonly object _lock;
+
+    public ConcurrentPacketAcceptanceStrategy(IPacketAcceptanceStrategy strategy)
+    {
+        _strategy = strategy;
+        _lock = new object();
+    }
+
+    public bool Accepts(ZeroPacket packet)
+    {
+        lock (_lock)
```

Comment:
Can we make the other strategy thread-safe with Interlock?

Reply:
Could you elaborate? You mean making `RateLimitedPacketAcceptanceStrategy` thread-safe directly using `Interlocked` operations?

Reply:
Probably should make `RateLimitedPacketAcceptanceStrategy` concurrent safe so you don't need this class.